### ISSUE TITLE
5.2: test suite is failing

### ISSUE THREAD

**Issue by kloczek** [pytest-dev/pytest-runner#56]
```
+ cd pytest-runner-5.2
+ tox --skip-missing-interpreters
.tox create: /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2/.tox/.tox
.tox installdeps: tox-pip-version>=0.0.6, tox-venv, tox >= 3.2
python create: /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2/.tox/python
python: pip_version is pip
python installdeps: setuptools>=31.0.1
python develop-inst: /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2
python installed: apipkg==1.5,appdirs==1.4.4,attrs==20.3.0,black==20.8b1,click==7.1.2,contextlib2==0.6.0.post1,coverage==5.3.1,distlib==0.3.1,docutils==0.16,execnet==1.7.1,filelock==3.0.12,flake8==3.8.4,iniconfig==1.1.1,mccabe==0.6.1,mock==4.0.3,more-itertools==8.6.0,mypy-extensions==0.4.3,packaging==20.8,path==15.0.1,path.py==12.5.0,pathspec==0.8.1,pluggy==0.13.1,py==1.10.0,pycodestyle==2.6.0,pyflakes==2.2.0,pyparsing==2.4.7,pytest==6.2.1,pytest-black==0.3.12,pytest-black-multipy==1.0.0,pytest-checkdocs==2.2.0,pytest-cov==2.10.1,pytest-fixture-config==1.7.0,pytest-flake8==1.0.7,# Editable install with no version control (pytest-runner==5.2),-e /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2,pytest-shutil==1.7.0,pytest-virtualenv==1.7.0,regex==2020.11.13,six==1.15.0,termcolor==1.1.0,toml==0.10.2,typed-ast==1.4.2,typing-extensions==3.7.4.3,virtualenv==20.2.2
python run-test-pre: PYTHONHASHSEED='270599559'
python run-test: commands[0] | pytest
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.3, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
cachedir: .tox/python/.pytest_cache
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2, configfile: pytest.ini
plugins: shutil-1.7.0, black-0.3.12, virtualenv-1.7.0, flake8-1.0.7, cov-2.10.1, checkdocs-2.2.0, black-multipy-1.0.0
collected 14 items

ptr.py .F.                                                                                                                                                           [ 21%]
setup.py ...                                                                                                                                                         [ 42%]
docs/conf.py ..                                                                                                                                                      [ 57%]
tests/test_ptr.py ......                                                                                                                                             [100%]

================================================================================= FAILURES =================================================================================
____________________________________________________________________________ Black format check ____________________________________________________________________________
--- /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2/ptr.py      2019-10-26 22:07:01 +0000
+++ /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2/ptr.py      2021-01-02 17:00:33.900912 +0000
@@ -36,12 +36,12 @@

     allow_hosts = None
     index_url = None

     def fetch_build_egg(self, req):
-        """ Specialized version of Distribution.fetch_build_egg
-        that respects respects allow_hosts and index_url. """
+        """Specialized version of Distribution.fetch_build_egg
+        that respects respects allow_hosts and index_url."""
         from setuptools.command.easy_install import easy_install

         dist = Distribution({'script_args': ['easy_install']})
         dist.parse_config_files()
         opts = dist.get_option_dict('easy_install')


----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                                                                                     Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------------------------------------------------------------
.tox/python/lib/python3.8/site-packages/appdirs.py                                                         257    205    20%   29-39, 78-86, 88-90, 96, 131-163, 196, 202, 236-254, 290-311, 345-353, 388-404, 411-415, 419, 424, 429, 434, 439, 444, 449, 460-476, 480-503, 507-530, 533-556, 559-571, 577-608
.tox/python/lib/python3.8/site-packages/distlib/__init__.py                                                  8      0   100%
.tox/python/lib/python3.8/site-packages/distlib/compat.py                                                   39      3    92%   474-475, 477
.tox/python/lib/python3.8/site-packages/distlib/resources.py                                               225    154    32%   30-33, 44, 53-69, 74-75, 93, 98-100, 104, 108, 116, 125, 130-132, 135, 140-147, 150, 153, 156-165, 168, 171-172, 175, 178-181, 184, 189-206, 214-222, 225, 228-243, 246-248, 251, 254, 257-258, 261-273, 276-284, 295-296, 300-301, 305, 316-332, 345-355
.tox/python/lib/python3.8/site-packages/distlib/scripts.py                                                 164     65    60%   57-63, 100, 139, 144, 153-155, 163-164, 174, 191, 214-215, 230-231, 237, 278-279, 288-291, 304, 310-359, 363, 367, 405, 416-419
.tox/python/lib/python3.8/site-packages/distlib/util.py                                                   1167    937    20%   65-142, 150-262, 269-289, 293-299, 312-315, 319-331, 335-341, 344-387, 391-406, 411-415, 419-424, 429-434, 444-449, 461-475, 491, 505-511, 516-528, 531-543, 549, 555, 563, 574-581, 584-598, 601-619, 622-631, 638-641, 644-659, 662-673, 685, 692-699, 712-714, 722, 725, 731, 735, 755-778, 791-795, 799-801, 805-816, 820-822, 825-832, 845-861, 876-880, 883-902, 908-925, 930-933, 936-938, 958-960, 966, 972-982, 990, 1002-1010, 1019-1022, 1029, 1042-1052, 1059-1061, 1064, 1067-1080, 1083-1085, 1088-1096, 1101, 1105-1127, 1132-1178, 1182-1190, 1201-1250, 1255-1264, 1277-1282, 1285-1292, 1295-1296, 1299-1300, 1303-1305, 1309, 1313-1320, 1323-1329, 1333-1348, 1352-1360, 1373-1379, 1383-1407, 1424-1455, 1463-1465, 1477-1481, 1484-1491, 1504, 1514-1526, 1531-1532, 1535-1543, 1548-1549, 1552-1564, 1569-1581, 1589-1596, 1607, 1610, 1615-1623, 1626, 1629-1634, 1640-1641, 1644-1651, 1663-1664, 1667-1695, 1698-1701, 1705-1709, 1717-1718, 1725-1739, 1742-1755, 1761
.tox/python/lib/python3.8/site-packages/filelock.py                                                        179     64    64%   39-40, 49-50, 57-58, 95-96, 99-101, 122, 125-126, 197, 203, 276-290, 345-358, 361-372, 388-389, 414-421, 424-432, 444, 448-451
.tox/python/lib/python3.8/site-packages/pytest_cov/embed.py                                                 77     66    14%   16-39, 53-79, 82-90, 97-137
.tox/python/lib/python3.8/site-packages/six.py                                                             493    239    52%   49-72, 98-99, 112, 118-121, 131-133, 145, 154-157, 192-193, 203, 222-223, 308, 488, 496, 501-507, 519-525, 530-532, 538-540, 545, 550, 554-568, 583, 586, 589, 592, 600-616, 628, 631, 645-647, 653-673, 679, 683, 687, 691, 698-721, 737-738, 743-795, 797-804, 814-834, 845-861, 870-873, 893-897, 914-920, 935, 939, 950-957, 978-979
.tox/python/lib/python3.8/site-packages/virtualenv/__init__.py                                               4      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/__main__.py                                              48     14    71%   18-22, 63-73
.tox/python/lib/python3.8/site-packages/virtualenv/activation/__init__.py                                    9      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/activation/activator.py                                  15      1    93%   44
.tox/python/lib/python3.8/site-packages/virtualenv/activation/bash/__init__.py                               8      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/activation/batch/__init__.py                             15      5    67%   16-18, 22-23
.tox/python/lib/python3.8/site-packages/virtualenv/activation/cshell/__init__.py                             9      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/activation/fish/__init__.py                               6      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/activation/powershell/__init__.py                         6      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/activation/python/__init__.py                            23      2    91%   33-34
.tox/python/lib/python3.8/site-packages/virtualenv/activation/via_template.py                               44      3    93%   16, 23, 30
.tox/python/lib/python3.8/site-packages/virtualenv/activation/xonsh/__init__.py                              9      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/app_data/__init__.py                                     33     10    70%   20, 28, 35, 38-42, 47-48
.tox/python/lib/python3.8/site-packages/virtualenv/app_data/base.py                                         64     16    75%   28, 32, 36, 40, 44, 48, 52, 58-59, 66, 71, 78, 82, 86, 90, 95
.tox/python/lib/python3.8/site-packages/virtualenv/app_data/na.py                                           36     10    72%   15, 26, 29, 32, 37, 41, 44, 52, 56, 66
.tox/python/lib/python3.8/site-packages/virtualenv/app_data/read_only.py                                    21      8    62%   12-14, 17, 20, 23, 26, 31
.tox/python/lib/python3.8/site-packages/virtualenv/app_data/via_disk_folder.py                             111     33    70%   56, 62-63, 70-72, 76-84, 95-101, 137-146, 149-150
.tox/python/lib/python3.8/site-packages/virtualenv/app_data/via_tempdir.py                                  17      5    71%   16-17, 23-24, 27
.tox/python/lib/python3.8/site-packages/virtualenv/config/__init__.py                                        1      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/config/cli/__init__.py                                    1      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/config/cli/parser.py                                     91     17    81%   21, 35, 39, 60, 80, 82-85, 87, 100, 115-120
.tox/python/lib/python3.8/site-packages/virtualenv/config/convert.py                                        52     27    48%   13, 16, 32-34, 39-41, 49-55, 63-74, 79-83
.tox/python/lib/python3.8/site-packages/virtualenv/config/env_var.py                                        15      7    53%   19-26
.tox/python/lib/python3.8/site-packages/virtualenv/config/ini.py                                            58     25    57%   37-38, 41-47, 49, 52-54, 57-69
.tox/python/lib/python3.8/site-packages/virtualenv/create/__init__.py                                        1      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/creator.py                                       131     37    72%   72, 106, 113-114, 128-130, 132, 140, 147, 150, 159, 168-169, 206-208, 212, 216-238
.tox/python/lib/python3.8/site-packages/virtualenv/create/describe.py                                       83      5    94%   55-57, 86, 89
.tox/python/lib/python3.8/site-packages/virtualenv/create/pyenv_cfg.py                                      47     14    70%   25-31, 44-45, 51, 54, 57-58, 61
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/__init__.py                         0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/api.py                             73      5    93%   22, 50, 54-55, 69
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/__init__.py                 0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/builtin_way.py             10      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/cpython/__init__.py         1      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/cpython/common.py          43      9    79%   48-53, 57, 62-64
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/cpython/cpython2.py        67     33    51%   21-26, 30, 34, 39, 43, 48-54, 63-70, 82-86, 94-102
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/cpython/cpython3.py        55     20    64%   29, 50-52, 56-60, 64, 68-72, 76-81, 84
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/cpython/mac_os.py         184    140    24%   29-34, 37-49, 53-58, 62, 66, 72, 75, 79-83, 87-103, 109, 112, 116-131, 158-168, 172-298
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/pypy/__init__.py            0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/pypy/common.py             36     15    58%   21-24, 28, 37-40, 45-49, 53
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/pypy/pypy2.py              77     32    58%   23, 27-32, 36, 40, 44, 49, 61, 64-71, 79, 87, 91, 95, 99-103, 111, 115, 119-121
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/pypy/pypy3.py              39     13    67%   18, 26, 30, 38, 41, 45-50, 59, 63
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/python2/__init__.py         0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/python2/python2.py         74     41    45%   25-51, 55, 59, 63-77, 81-86, 90-93, 96, 99, 103, 107, 111
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/ref.py                    127     30    76%   45-46, 52, 61-64, 71, 78-83, 87, 91, 93, 109, 120, 132-138, 148, 159, 163, 167, 169, 172
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/builtin/via_global_self_do.py      81     13    84%   42, 44-52, 54, 59, 76, 100
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/store.py                           10      5    50%   7-9, 13-14
.tox/python/lib/python3.8/site-packages/virtualenv/create/via_global_ref/venv.py                            60     36    40%   17-22, 25, 32, 34, 37-43, 46-54, 57-61, 64-69, 73-75, 78-83
.tox/python/lib/python3.8/site-packages/virtualenv/discovery/__init__.py                                     1      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/discovery/builtin.py                                    116     68    41%   40, 43, 46-47, 57, 62, 70-72, 75-101, 105-115, 120-121, 124, 127-139, 143-151, 156-159
.tox/python/lib/python3.8/site-packages/virtualenv/discovery/cached_py_info.py                              96     49    49%   30-34, 43, 57-58, 60, 69, 71-76, 81-113, 118-119, 122-130, 133-136, 140-141
.tox/python/lib/python3.8/site-packages/virtualenv/discovery/discover.py                                    20      2    90%   18, 36
.tox/python/lib/python3.8/site-packages/virtualenv/discovery/py_info.py                                    326    168    48%   45, 70-71, 97, 115, 119, 136, 144-145, 149, 153, 158, 166-170, 174-183, 194-197, 200, 206-235, 244-247, 254-275, 286-288, 296-298, 302, 305-308, 321-325, 331-332, 346-358, 360, 367-389, 392-411, 417-434, 437-452, 455-463, 466-484, 490
.tox/python/lib/python3.8/site-packages/virtualenv/discovery/py_spec.py                                     90     60    33%   34-67, 72-91, 95, 99-109, 112, 122
.tox/python/lib/python3.8/site-packages/virtualenv/info.py                                                  40      9    78%   39-47
.tox/python/lib/python3.8/site-packages/virtualenv/report.py                                                34      1    97%   50
.tox/python/lib/python3.8/site-packages/virtualenv/run/__init__.py                                          71      5    93%   67, 82, 87-88, 118
.tox/python/lib/python3.8/site-packages/virtualenv/run/plugin/__init__.py                                    0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/run/plugin/activators.py                                 28      1    96%   34
.tox/python/lib/python3.8/site-packages/virtualenv/run/plugin/base.py                                       44      3    93%   9, 43, 48
.tox/python/lib/python3.8/site-packages/virtualenv/run/plugin/creators.py                                   50      7    86%   24, 28, 39-43, 76
.tox/python/lib/python3.8/site-packages/virtualenv/run/plugin/discovery.py                                  15      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/run/plugin/seeders.py                                    16      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/run/session.py                                           58      4    93%   23, 28, 88, 91
.tox/python/lib/python3.8/site-packages/virtualenv/seed/__init__.py                                          1      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/__init__.py                                    0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/base_embed.py                                 53      3    94%   35, 106, 110
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/pip_invoke.py                                 40     26    35%   15, 18-23, 27-32, 36-56
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/via_app_data/__init__.py                       0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/via_app_data/pip_install/__init__.py           0      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py             121      7    94%   30, 38-41, 108, 139-140
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/via_app_data/pip_install/copy.py              26      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/via_app_data/pip_install/symlink.py           44     30    32%   16-18, 22-45, 48-51, 54-56, 59-61
.tox/python/lib/python3.8/site-packages/virtualenv/seed/embed/via_app_data/via_app_data.py                 101     27    73%   41, 58-59, 67-71, 95-97, 99-112, 126, 132-133
.tox/python/lib/python3.8/site-packages/virtualenv/seed/seeder.py                                           13      2    85%   30, 39
.tox/python/lib/python3.8/site-packages/virtualenv/seed/wheels/__init__.py                                   4      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/wheels/acquire.py                                   59     15    75%   27, 64, 80, 84-96, 118
.tox/python/lib/python3.8/site-packages/virtualenv/seed/wheels/bundle.py                                    31      7    77%   23-26, 38, 47-48
.tox/python/lib/python3.8/site-packages/virtualenv/seed/wheels/embed/__init__.py                            10      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/seed/wheels/periodic_update.py                          209     39    81%   33-36, 50, 115, 120, 137, 162, 192, 202, 241, 253, 269-271, 277, 297-300, 305-315, 319-354
.tox/python/lib/python3.8/site-packages/virtualenv/seed/wheels/util.py                                      81     32    60%   20, 40-41, 43, 51-75, 85-92, 111, 115-116
.tox/python/lib/python3.8/site-packages/virtualenv/util/__init__.py                                          6      1    83%   8
.tox/python/lib/python3.8/site-packages/virtualenv/util/error.py                                             8      5    38%   9-13
.tox/python/lib/python3.8/site-packages/virtualenv/util/lock.py                                            122     22    82%   20-23, 52, 62, 66, 71, 76, 103-104, 107, 119-124, 150, 153, 157, 161
.tox/python/lib/python3.8/site-packages/virtualenv/util/path/__init__.py                                     5      0   100%
.tox/python/lib/python3.8/site-packages/virtualenv/util/path/_pathlib/__init__.py                           33     26    21%   12-60
.tox/python/lib/python3.8/site-packages/virtualenv/util/path/_permission.py                                 21     14    33%   10-20, 24-26
.tox/python/lib/python3.8/site-packages/virtualenv/util/path/_sync.py                                       59     13    78%   15-16, 31, 34-39, 69-73, 84
.tox/python/lib/python3.8/site-packages/virtualenv/util/six.py                                              22     16    27%   12-28, 33-50
.tox/python/lib/python3.8/site-packages/virtualenv/util/subprocess/__init__.py                              18      9    50%   9-11, 20-32
.tox/python/lib/python3.8/site-packages/virtualenv/util/zipapp.py                                           24     15    38%   12-15, 19-24, 28-33
.tox/python/lib/python3.8/site-packages/virtualenv/version.py                                                2      0   100%
docs/conf.py                                                                                                 3      0   100%
ptr.py                                                                                                     103     61    41%   26-32, 43-84, 133-134, 142, 152, 161-175, 179-187, 194-209, 213, 219-222
tests/test_ptr.py                                                                                           56      0   100%
--------------------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                     6784   3111    54%

========================================================================= short test summary info ==========================================================================
FAILED ptr.py::BLACK
====================================================================== 1 failed, 13 passed in 35.91s =======================================================================
ERROR: InvocationError for command /home/tkloczko/rpmbuild/BUILD/pytest-runner-5.2/.tox/python/bin/pytest (exited with code 1)
_________________________________________________________________________________ summary __________________________________________________________________________________
ERROR:   python: commands failed
```

**Comment by RonnyPfannschmidt** [comment]
This looks like the result of a insufficient pin of black

**Comment by kloczek** [comment]
Really sorry but your reply says nothing to me :P

**Comment by RonnyPfannschmidt** [comment]
the tox build has a dependency on black (a code format tool) thats not pinned tightly enough, black recently changed output, thus the tests break with the slightly changed output

**Comment by graingert** [comment]
> pytest-runner depends on deprecated features of setuptools and relies on features that break security mechanisms in pip. 

I think archiving the project would make the most sense here

**Comment by jaraco** [comment]
Fixed in f11649c7d89d69bdd75b4e5111c6528c381d3c29.
